fish-pepper/run-java-sh/fp-files/java-default-options

```diff
@@ -111,43 +111,19 @@ diagnostic_options() {
   fi
 }
 
-add2() {
-  echo "$1" "$2" | awk '{printf "%d\n", $1+$2}'
-}
-
-div2() {
-  echo "$1" "$2" | awk '{printf "%d\n", $1/$2}'
-}
-
-mul2() {
-  echo "$1" "$2" | awk '{printf "%d\n", $1*$2}'
-}
-
-sub2() {
-  echo "$1" "$2" | awk '{printf "%d\n", $1-$2}'
-}
-
-log2_int() {
-  echo "$1" | awk '{printf "%d\n", log($1)/log(2)}'
-}
-
-max2() {
-  [ "$1" -le "$2" ] && echo "$2" || echo "$1"
-}
-
 # Replicate thread ergonomics for tiered compilation.
 # This could ideally be skipped when tiered compilation is disabled.
 # The algorithm is taken from:
 # OpenJDK / jdk8u / jdk8u / hotspot
 # src/share/vm/runtime/advancedThresholdPolicy.cpp
 ci_compiler_count() {
   local core_limit="$1"
-  local log_cpu=$(log2_int "$core_limit")
-  local loglog_cpu=$(log2_int $(max2 "$log_cpu" 1))
-  local count=$(mul2 $(max2 $(mul2 "$log_cpu" "$loglog_cpu") 1) 1.5)
-  local c1_count=$(max2 $(div2 "$count" 3) 1)
-  local c2_count=$(max2 $(sub2 "$count" "$c1_count") 1)
-  [ $(c2_disabled) = true ] && echo "$c1_count" || echo $(add2 "$c1_count" "$c2_count")
+  local log_cpu=$(int "$core_limit" "log2(\$1)")
+  local loglog_cpu=$(int "$log_cpu" "log2(max2(\$1,1))")
+  local count=$(int "$log_cpu" "$loglog_cpu" "max2(\$1*\$2,1)*3/2")
+  local c1_count=$(int "$count" "max2(\$1/3,1)")
+  local c2_count=$(int "$count" "$c1_count" "max2(\$1-\$2,1)")
```

Comment:
If you use `'` instead of `"` you can avoid escaping which makes it a bit easier to read.

Another suggestion: What about putting the formula first and the arguments later ? feels a bit more natural for people using varargs:

Suggestion: 

```
<formula> <result format> <arg1> <arg2> ...
````

Reply:
Make sense. I did that first and then change my mind. Let me do as you suggest.